aves/utils.py

```diff
@@ -0,0 +1,101 @@
+"""Utility functions for AVES."""
+
+import os
+from pathlib import Path
+
+import numpy as np
+import torch
+import torchaudio
+
+ALLOWED_AUDIO_FILE_EXTENSIONS = ["wav", "mp3", "flac", "ogg", "m4a"]
+TARGET_SR = 16000  # AVES works with 16kHz audio
+DEFAULT_DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
+
+
+def parse_audio_file_paths(audio_dir: str, audio_file_extension: str | None = None) -> list[Path]:
+    """Parse audio file paths from a directory or a list of paths.
+
+    Args:
+        audio_dir (str): Path to the directory containing the audio files.
+        audio_file_extension (str, optional): Extension of the audio files to process. Defaults to None.
+
+    Returns:
+        list[Path]: List of Path objects for the audio files.
+    """
+    audio_dir = Path(audio_dir)
+    if audio_file_extension is None:
+        audio_files = []
+        print(f"Searching for audio files in {audio_dir} with extensions: {ALLOWED_AUDIO_FILE_EXTENSIONS}")
+        for ext in ALLOWED_AUDIO_FILE_EXTENSIONS:
```

Comment:
you can do a torchaudio.info to see if the extension is supported by the backend.